src/main/java/se/sundsvall/supportmanagement/service/mapper/CommunicationMapper.java

```diff
@@ -0,0 +1,58 @@
+package se.sundsvall.supportmanagement.service.mapper;
+
+import static java.util.Collections.emptyList;
+
+import java.util.List;
+
+import se.sundsvall.supportmanagement.api.model.communication.Communication;
+import se.sundsvall.supportmanagement.api.model.communication.CommunicationAttachment;
+import se.sundsvall.supportmanagement.integration.db.model.CommunicationAttachmentEntity;
+import se.sundsvall.supportmanagement.integration.db.model.CommunicationEntity;
+
+public final class CommunicationMapper {
+
+	private CommunicationMapper() {
+		// Intentionally empty
+	}
+
+	public static List<Communication> toCommunications(final List<CommunicationEntity> entities) {
+		return entities.stream()
+			.map(CommunicationMapper::toCommunication)
+			.toList();
+	}
+
+	public static Communication toCommunication(final CommunicationEntity entity) {
+		return Communication.create()
+			.withCommunicationID(entity.getCommunicationID())
+			.withErrandNumber(entity.getErrandNumber())
+			.withDirection(entity.getDirection())
+			.withMessageBody(entity.getMessageBody())
+			.withSent(entity.getSent())
+			.withSubject(entity.getSubject())
+			.withCommunicationType(entity.getCommunicationType())
+			.withMobileNumber(entity.getMobileNumber())
+			.withEmail(entity.getEmail())
+			.withViewed(entity.isViewed())
+			.withCommunicationAttachments(toAttachments(entity.getAttachments()));
+	}
+
+	public static List<CommunicationAttachment> toAttachments(final List<CommunicationAttachmentEntity> attachments) {
+
+		if (attachments == null) {
+			return emptyList();
+		}
+
+		return attachments.stream()
```

Comment:
If you use Optional.ofNullable here, you can skip the if-clause above and get higher coverage with less tests.